samples/WinForms/MainFormWinFormsDx12.cs

```diff
@@ -0,0 +1,39 @@
+// Copyright © Tanner Gooding and Contributors. Licensed under the MIT License (MIT). See License.md in the repository root for more information.
+
+using System;
+using System.Windows.Forms;
+
+namespace TerraFX.Samples.WinForms
+{
+    /// <summary>
+    /// Main form of the project TerraFX.Samples.WinForms.
+    /// It is a WinForms GUI with a Panel that contains DirectX content.
+    /// The DirectX content is taken from the Hello Triangle example.
+    /// The main point is to demonstrate how to connect a WinForms GUI to DirectX rendering.
+    /// This connection is encapsulated in the DX12Panel class.
+    /// </summary>
+    public partial class MainFormWinFormsDX12 : Form
+    {
+        private readonly DX12Panel _panel;
+
+        public MainFormWinFormsDX12()
+        {
+            InitializeComponent();
+            _panel = InitDX12Panel();
+        }
+
+        private DX12Panel InitDX12Panel()
```

Comment:
Shouldn't this all be done in the designer?

Reply:
The designer does not let me use the custom panel :(

Reply:
What is the error it gives?

Reply:
It needed to be a separate project and I think it did not like the .Net 5.0 preview. But I need to find the instructions again on making a custom panel ...

Reply:
I was following https://docs.microsoft.com/en-us/dotnet/framework/winforms/controls/creating-a-wf-control-design-time-features
Will try again.

Reply:
So what got me off the track of making a designer enabled panel was that it requires a new project and even Visual Studio 2019 Preview has no option for WinForms control library plus .Net Core or .Net 5.0. So the project file style and the target framework both do not match.  So I stayed with adding the panel at runtime.

Reply:
It shouldn't require a new project, but it does look like the designer for .NET Core doesn't support custom controls yet and you'll get an error if you try to drag/drop.

Reply:
Right. So for now it has to be added at runtime. Which means not great, but nothing to change for now.